[CI] GetResultTests.testToAndFromXContentEmbedded failure

### CI Link

https://gradle-enterprise.elastic.co/s/m2gyt6k4isde4/tests/overview?outcome=FAILED

### Repro line

./gradlew ":server:test" --tests "org.elasticsearch.index.get.GetResultTests.testToAndFromXContentEmbedded" -Dtests.seed=764C9696901B0FA8 -Dtests.locale=es-PR -Dtests.timezone=Australia/Perth -Druntime.java=22

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

```
2> REPRODUCE WITH: ./gradlew ":server:test" --tests "org.elasticsearch.index.get.GetResultTests.testToAndFromXContentEmbedded" -Dtests.seed=764C9696901B0FA8 -Dtests.locale=es-PR -Dtests.timezone=Australia/Perth -Druntime.java=22
  2> java.lang.AssertionError: expected: org.elasticsearch.index.get.GetResult<{
      "_index" : null,
      "_id" : null,
      "_seq_no" : 6347641838303503358,
      "_primary_term" : 34,
      "_doc_count" : "PHjdZV",
      "_ignored_source" : "PQUHluJ",
      "found" : true,
      "_source" : {
        "DtMbkgoe" : 6439978940484883701,
        "mnaVsyWVq" : "dpsrVTdQQD"
      },
      "fields" : {
        "YRikh" : [
          [
            -1244118694
          ],
          [
            -965887493,
            1718301905,
            994396153,
            -1363971690
          ],
          [
            1374939399,
            -549215428,
            1648860293,
            1872283781,
            521072097
          ],
          [
            -209300000,
            -1110505785
          ]
        ],
        "sIZeU" : [
          {
            "vZCny" : "YSABehkEJR",
            "NWJYs" : false,
            "IHkgy" : 1489359398
          },
          {
            "tfQTX" : -495041849,
            "DsRPR" : false,
            "gcQxR" : "dBCaYXlJgU"
          },
          {
            "iazas" : "sibTJnJpPT",
            "QgVZZ" : -1963391815,
            "wfHEn" : true
          }
        ],
        "NHvEd" : [
          {
            "djWPq" : "KbzNsNYMBA",
            "lMzNx" : false,
            "UDNdK" : -498254522
          },
          {
            "qxroT" : "FFcTpjPxmI",
            "VhLaT" : true,
            "adkJk" : -1214480544
          },
          {
            "udrcm" : -1154896101,
            "xGoCQ" : "jyrjmcopxq",
            "hioPo" : false
          }
        ],
        "LAhDU" : [
          [
            -1668094594,
            1247158066,
            1205769889
          ],
          [
            678151574,
            968808396,
            -1207595562,
            928609448,
            -1746758814
          ],
          [
            -671079043,
            1803093487,
            1081872460,
            1806171758
          ],
          [
            -368121274,
            1655840897,
            391451726
          ]
        ],
        "TsrMY" : [
          [
            1799275917,
            492320020,
            -843631093,
            1785857873,
            -1767124891
          ],
          [
            -1431980803,
            -1823861119,
            -694547500,
            1219608223
          ],
          [
            -2059393207,
            -445653418
          ],
          [
            -48112774,
            1167691112,
            460753959,
            1934140366,
            -616250120
          ],
          [
            -2076557883,
            -1561045622,
            -1243457171,
            626064619
          ]
        ],
        "ipaDP" : [
          [
            -1078450373,
            -1058939401,
            -715016887,
            344380702,
            -1846475521
          ],
          [
            -1427926825,
            1270109173,
            870438100,
            154513818
          ]
        ],
        "ERPAm" : [
          [
            1387991381,
            -2130787691
          ],
          [
            -380768858,
            -1391934665,
            1641138638,
            -1127506243,
            1451113048
          ]
        ],
        "eirXW" : [
          [
            2112631908,
            1016544991,
            -940106333,
            120523577
          ],
          [
            -457924078,
            -1337529229,
            -1454271552
          ]
        ],
        "laSce" : [
          [
            -1206048835,
            -1744569988,
            -761778537,
            994171977
          ],
          [
            2123457533
          ],
          [
            1377706965,
            -1300750629
          ],
          [
            1621881543,
            -1173926105,
            1755728824,
            1426660418,
            -865164462
          ],
          [
            -1486861101,
            1491277714,
            -947710875
          ]
        ]
      }
    }> but was: org.elasticsearch.index.get.GetResult<{
      "_index" : null,
      "_id" : null,
      "_seq_no" : 6347641838303503358,
      "_primary_term" : 34,
      "_doc_count" : "PHjdZV",
      "_ignored_source" : "PQUHluJ",
      "found" : true,
      "_source" : {
        "DtMbkgoe" : 6439978940484883701,
        "mnaVsyWVq" : "dpsrVTdQQD"
      },
      "fields" : {
        "YRikh" : [
          [
            -1244118694
          ],
          [
            -965887493,
            1718301905,
            994396153,
            -1363971690
          ],
          [
            1374939399,
            -549215428,
            1648860293,
            1872283781,
            521072097
          ],
          [
            -209300000,
            -1110505785
          ]
        ],
        "sIZeU" : [
          {
            "vZCny" : "YSABehkEJR",
            "NWJYs" : false,
            "IHkgy" : 1489359398
          },
          {
            "tfQTX" : -495041849,
            "DsRPR" : false,
            "gcQxR" : "dBCaYXlJgU"
          },
          {
            "iazas" : "sibTJnJpPT",
            "QgVZZ" : -1963391815,
            "wfHEn" : true
          }
        ],
        "NHvEd" : [
          {
            "djWPq" : "KbzNsNYMBA",
            "lMzNx" : false,
            "UDNdK" : -498254522
          },
          {
            "qxroT" : "FFcTpjPxmI",
            "VhLaT" : true,
            "adkJk" : -1214480544
          },
          {
            "udrcm" : -1154896101,
            "xGoCQ" : "jyrjmcopxq",
            "hioPo" : false
          }
        ],
        "LAhDU" : [
          [
            -1668094594,
            1247158066,
            1205769889
          ],
          [
            678151574,
            968808396,
            -1207595562,
            928609448,
            -1746758814
          ],
          [
            -671079043,
            1803093487,
            1081872460,
            1806171758
          ],
          [
            -368121274,
            1655840897,
            391451726
          ]
        ],
        "TsrMY" : [
          [
            1799275917,
            492320020,
            -843631093,
            1785857873,
            -1767124891
          ],
          [
            -1431980803,
            -1823861119,
            -694547500,
            1219608223
          ],
          [
            -2059393207,
            -445653418
          ],
          [
            -48112774,
            1167691112,
            460753959,
            1934140366,
            -616250120
          ],
          [
            -2076557883,
            -1561045622,
            -1243457171,
            626064619
          ]
        ],
        "ipaDP" : [
          [
            -1078450373,
            -1058939401,
            -715016887,
            344380702,
            -1846475521
          ],
          [
            -1427926825,
            1270109173,
            870438100,
            154513818
          ]
        ],
        "ERPAm" : [
          [
            1387991381,
            -2130787691
          ],
          [
            -380768858,
            -1391934665,
            1641138638,
            -1127506243,
            1451113048
          ]
        ],
        "eirXW" : [
          [
            2112631908,
            1016544991,
            -940106333,
            120523577
          ],
          [
            -457924078,
            -1337529229,
            -1454271552
          ]
        ],
        "laSce" : [
          [
            -1206048835,
            -1744569988,
            -761778537,
            994171977
          ],
          [
            2123457533
          ],
          [
            1377706965,
            -1300750629
          ],
          [
            1621881543,
            -1173926105,
            1755728824,
            1426660418,
            -865164462
          ],
          [
            -1486861101,
            1491277714,
            -947710875
          ]
        ]
      }
    }>
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -251,18 +251,12 @@ tests:
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/115631
-- class: org.elasticsearch.index.get.GetResultTests
-  method: testToAndFromXContent
-  issue: https://github.com/elastic/elasticsearch/issues/115688
 - class: org.elasticsearch.action.update.UpdateResponseTests
   method: testToAndFromXContent
   issue: https://github.com/elastic/elasticsearch/issues/115689
 - class: org.elasticsearch.xpack.shutdown.NodeShutdownIT
   method: testStalledShardMigrationProperlyDetected
   issue: https://github.com/elastic/elasticsearch/issues/115697
-- class: org.elasticsearch.index.get.GetResultTests
-  method: testToAndFromXContentEmbedded
-  issue: https://github.com/elastic/elasticsearch/issues/115657
 - class: org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT
   method: testGeoShapeGeoHash
   issue: https://github.com/elastic/elasticsearch/issues/115664

--- a/server/src/main/java/org/elasticsearch/index/get/GetResult.java
+++ b/server/src/main/java/org/elasticsearch/index/get/GetResult.java
@@ -21,6 +21,7 @@ import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.logging.DeprecationLogger;
 import org.elasticsearch.common.xcontent.XContentHelper;
 import org.elasticsearch.index.mapper.IgnoredFieldMapper;
+import org.elasticsearch.index.mapper.IgnoredSourceFieldMapper;
 import org.elasticsearch.index.mapper.MapperService;
 import org.elasticsearch.index.mapper.SourceFieldMapper;
 import org.elasticsearch.search.lookup.Source;
@@ -247,7 +248,7 @@ public class GetResult implements Writeable, Iterable<DocumentField>, ToXContent
 
         for (DocumentField field : metaFields.values()) {
             // TODO: can we avoid having an exception here?
-            if (field.getName().equals(IgnoredFieldMapper.NAME)) {
+            if (field.getName().equals(IgnoredFieldMapper.NAME) || field.getName().equals(IgnoredSourceFieldMapper.NAME)) {
                 builder.field(field.getName(), field.getValues());
             } else {
                 builder.field(field.getName(), field.<Object>getValue());
@@ -341,7 +342,7 @@ public class GetResult implements Writeable, Iterable<DocumentField>, ToXContent
                     parser.skipChildren(); // skip potential inner objects for forward compatibility
                 }
             } else if (token == XContentParser.Token.START_ARRAY) {
-                if (IgnoredFieldMapper.NAME.equals(currentFieldName)) {
+                if (IgnoredFieldMapper.NAME.equals(currentFieldName) || IgnoredSourceFieldMapper.NAME.equals(currentFieldName)) {
                     metaFields.put(currentFieldName, new DocumentField(currentFieldName, parser.list()));
                 } else {
                     parser.skipChildren(); // skip potential inner arrays for forward compatibility

